crates/uv-client/src/registry_client.rs

```diff
@@ -88,17 +88,17 @@ impl RegistryClientBuilder {
 
     pub fn build(self) -> RegistryClient {
         let client_raw = self.client.unwrap_or_else(|| {
-            // Get pip timeout from env var
+            // Timeout options, matching https://doc.rust-lang.org/nightly/cargo/reference/config.html#httptimeout
+            // `UV_REQUEST_TIMEOUT` is provided for backwards compatibility
             let default_timeout = 5 * 60;
-            let timeout = env::var("UV_REQUEST_TIMEOUT")
-                .map_err(|_| default_timeout)
-                .and_then(|value| {
-                    value.parse::<u64>()
-                        .map_err(|_| {
-                            warn_user_once!("Ignoring invalid value for UV_REQUEST_TIMEOUT. Expected integer number of seconds, got {value}.");
-                            default_timeout
-                        })
-                }).unwrap_or(default_timeout);
+            let timeout = env::var("UV_REQUEST_TIMEOUT").or_else(|_| env::var("UV_HTTP_TIMEOUT")).or_else(|_| env::var("HTTP_TIMEOUT")).and_then(|value| {
```

Comment:
Should `UV_REQUEST_TIMEOUT` be last? I can't tell.

Reply:
Probably before `HTTP_REQUEST` but after `UV_HTTP_TIMEOUT`? Unclear. Wish I could drop support for it haha.